test/integration/backward_compatible/HazelcastClientTest.js

```diff
@@ -116,44 +116,31 @@ configParams.forEach(function (cfg) {
             expect(info.labels).to.deep.equal(new Set());
         });
 
-        it('getDistributedObjects returns all dist objects', function (done) {
+        it('getDistributedObjects returns all dist objects', function () {
             managed.getObject(client.getMap.bind(client, 'map'));
             managed.getObject(client.getSet.bind(client, 'set'));
-            setTimeout(function () {
-                client.getDistributedObjects().then(function (distObjects) {
-                    try {
-                        const names = distObjects.map((o) => {
-                            return o.getName();
-                        });
-                        expect(names).to.have.members(['map', 'set']);
-                        done();
-                    } catch (e) {
-                        done(e);
-                    }
+            return TestUtil.assertTrueEventually(async function () {
+                const distObjects = await client.getDistributedObjects();
+                const names = distObjects.map((o) => {
+                    return o.getName();
                 });
-            }, 300);
+                expect(names).to.have.members(['map', 'set']);
+            }, 100, 5000);
         });
 
-        it('getDistributedObjects does not return removed object', function (done) {
+        it('getDistributedObjects does not return removed object', function () {
             managed.getObject(client.getMap.bind(client, 'map1'));
             managed.getObject(client.getMap.bind(client, 'map2'));
             managed.getObject(client.getMap.bind(client, 'map3'));
 
-            setTimeout(function () {
-                managed.destroy('map1').then(function () {
-                    client.getDistributedObjects().then(function (distObjects) {
-                        try {
-                            const names = distObjects.map(function (o) {
-                                return o.getName();
-                            });
-                            expect(names).to.have.members(['map2', 'map3']);
-                            done();
-                        } catch (e) {
-                            done(e);
-                        }
-                    });
+            return TestUtil.assertTrueEventually( async function () {
+                await managed.destroy('map1');
+                const distObjects = await client.getDistributedObjects();
+                const names = distObjects.map(function (o) {
```

Comment:
Same comment as above for arrow functions

Reply:
would we want to enable https://eslint.org/docs/rules/prefer-arrow-callback flag

Reply:
we can do that just for tests

Reply:
seems there is a mocha friendly one too https://github.com/lo1tuma/eslint-plugin-mocha/blob/master/docs/rules/prefer-arrow-callback.md

Reply:
we can add 

```
"mocha/prefer-arrow-callback": ["error", {
            "allowNamedFunctions": true
}]
```

to the .eslintrc in the test folder. This will allow named functions, mocha hooks. By default it won't change the function usage if it's behaviour would be different after the change.

Reply:
Sounds good to me. It would be good to be consistent across all the tests